src/Behat/Behat/Context/Attribute/AttributeReader.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+/*
+ * This file is part of the Behat.
+ * (c) Konstantin Kudryashov <ever.zet@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Behat\Behat\Context\Attribute;
+
+use ReflectionMethod;
+
+/**
+ * Reads Attributes of a provided context method into a Callee.
+ *
+ * @see AttributedContextReader
+ *
+ * @author Konstantin Kudryashov <ever.zet@gmail.com>
+ */
+interface AttributeReader
+{
+    /**
+     * Reads all callees associated with a provided method.
+     *
+     * @param string           $contextClass
+     * @param ReflectionMethod $method
+     *
+     * @return array
+     */
+    public function readCallees($contextClass, ReflectionMethod $method);
```

Comment:
shouldn't it be safe to add a type here ? And how about a generator instead ?

Reply:
(Because IIRC, the project was bumped to 7.2+)

Reply:
Yes that would work 👍 

I don't see a point in using a `Generator` though, given that is only called once and then stored in a caching layer for future calls anyway.

Reply:
Oh if it's stored it's indeed useless to use a generator. So :+1:

Reply:
I've added the `string` typehint